src/Psalm/Internal/Provider/ReturnTypeProvider/GetObjectVarsReturnTypeProvider.php

```diff
@@ -126,7 +142,11 @@ public static function getGetObjectVarsReturnType(
                 return new TKeyedArray(
                     $properties,
                     null,
-                    $class_storage->final ? null : [Type::getString(), Type::getMixed()],
+                    $class_storage->final
+                        || $class_storage->name === UnitEnum::class
```

Comment:
Annoying this is not covered by `interfaceExtends()` but I guess it makes sense.